File: backend/internal/model/card.go
package model

import "time"

type Card struct {
	ID          int64    `json:"id"`
	CardNumber  string   `json:"card_number" binding:"required"`
	Balance     float64  `json:"balance"`
	IsBlocked   bool     `json:"is_blocked"`
	OwnerName   string   `json:"owner_name"`
	KeyID       *int64   `json:"key_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateCardRequest struct {
	CardNumber string  `json:"card_number" binding:"required"`
	Balance    float64 `json:"balance"`
	IsBlocked  bool    `json:"is_blocked"`
	OwnerName  string  `json:"owner_name"`
	KeyID      *int64  `json:"key_id"`
}

type UpdateCardRequest struct {
	CardNumber string  `json:"card_number,omitempty"`
	Balance    float64 `json:"balance,omitempty"`
	IsBlocked  *bool   `json:"is_blocked,omitempty"`
	OwnerName  string  `json:"owner_name,omitempty"`
	KeyID      *int64  `json:"key_id,omitempty"`
}

File: frontend/src/commonMain/kotlin/com/transportpay/data/models/Card.kt
package com.transportpay.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Card(
    val id: Int,
    @SerialName("card_number") val cardNumber: String,
    val balance: Double,
    @SerialName("is_blocked") val isBlocked: Boolean,
    @SerialName("owner_name") val ownerName: String?,
    @SerialName("key_id") val keyId: Int?,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String
)

@Serializable
data class CreateCardRequest(
    @SerialName("card_number") val cardNumber: String,
    val balance: Double = 0.0,
    @SerialName("owner_name") val ownerName: String? = null,
    @SerialName("key_id") val keyId: Int? = null
)

@Serializable
data class UpdateCardRequest(
    val balance: Double? = null,
    @SerialName("is_blocked") val isBlocked: Boolean? = null,
    @SerialName("owner_name") val ownerName: String? = null,
    @SerialName("key_id") val keyId: Int? = null
)
